app/screens/custom_status/custom_status_modal.tsx

```diff
@@ -0,0 +1,315 @@
+
+        this.rightButton.text = props.intl.formatMessage({id: 'mobile.custom_status.modal_confirm', defaultMessage: 'Done'});
+        this.rightButton.color = props.theme.sidebarHeaderTextColor;
+
+        const buttons = {
+            rightButtons: [this.rightButton],
+        };
+
+        const options = {
+            topBar: {
+                title: {
+                    text: props.intl.formatMessage({id: 'custom_status.set_status', defaultMessage: 'Set a Status'}),
+                    alignment: 'center',
+                },
+                ...buttons,
+            },
+        };
+
+        mergeNavigationOptions(props.componentId, options);
+
+        this.state = {
+            emoji: props.customStatus.emoji,
+            text: props.customStatus.text || '',
+        };
+    }
+
+    navigationButtonPressed() {
+        this.handleSetStatus();
+    }
+
+    handleSetStatus = () => {
+        const {emoji, text} = this.state;
+        const customStatus = {
+            emoji: emoji || 'speech_balloon',
+            text: text.trim(),
+        };
+        this.props.actions.setCustomStatus(customStatus);
+        dismissModal();
+    };
+
+    handleTextChange = (value: string) => this.setState({text: value});
+
+    handleRecentCustomStatusClear = (status: UserCustomStatus) => this.props.actions.removeRecentCustomStatus(status);
+
+    clearHandle = () => {
+        this.setState({emoji: '', text: ''});
+    };
+
+    handleSuggestionClick = (status: UserCustomStatus) => {
+        const {emoji, text} = status;
+        this.setState({emoji, text});
+    };
+
+    renderRecentCustomStatuses = () => {
+        const {recentCustomStatuses, theme} = this.props;
+        const style = getStyleSheet(theme);
+        const recentStatuses = recentCustomStatuses.map((status: UserCustomStatus, index: number) => (
+            <CustomStatusSuggestion
+                key={status.text}
+                handleSuggestionClick={this.handleSuggestionClick}
+                handleClear={this.handleRecentCustomStatusClear}
+                emoji={status.emoji}
+                text={status.text}
+                theme={theme}
+                separator={index !== recentCustomStatuses.length - 1}
+            />
+        ));
+
+        if (recentStatuses.length <= 0) {
+            return null;
+        }
+
+        return (
+            <>
+                <View style={style.separator}/>
+                <View >
+                    <Text style={style.title}>
+                        {this.props.intl.formatMessage({
+                            id: 'custom_status.suggestions.recent_title',
+                            defaultMessage: 'RECENT',
+                        })}
+                    </Text>
+                    <View style={style.block}>
+                        {recentStatuses}
+                    </View>
+                </View >
+            </>
+        );
+    };
+
+    renderCustomStatusSuggestions = () => {
```

Comment:
maybe we can use one renderSuggestions() to render recent as well as custom status suggestions.